Open-Rival/src/Rival.cpp

```diff
@@ -209,12 +214,89 @@ namespace Rival {
     void Rival::update() {
 
         mousePicker->handleMouse();
+
         //std::cout << mousePicker->getTileX() << ", " << mousePicker->getTileY() << "\n";
 
         std::map<int, std::unique_ptr<Unit>>& units = scenario->getUnits();
         for (auto const& kv : units) {
             const std::unique_ptr<Unit>& unit = kv.second;
             //unit->rotateRight();
+            unit->tick();
+
+            // Test for animation subsystem
+            static const double randMax = RAND_MAX;
+            auto randGen = [&]() {
+                double r = static_cast<double>( rand() );
+                return r / randMax;
+            };
+            auto r = randGen();
+            if ( UnitType::Peasant == unit->getType() && r > 0.999 ) {
+                auto r = randGen();
+                if ( r <= 1.0 / 6.0 ) {
+                    unit->setAnimation( UnitAnimationType::Standing );
+                } else if ( r <= 2.0 / 6.0 ) {
+                    unit->setAnimation( UnitAnimationType::HoldingBag );
+                } else if ( r <= 3.0 / 6.0 ) {
+                    unit->setAnimation( UnitAnimationType::Moving );
+                } else if ( r <= 4.0 / 6.0 ) {
+                    unit->setAnimation( UnitAnimationType::Moving2 );
+                } else if ( r <= 5.0 / 6.0 ) {
+                    unit->setAnimation( UnitAnimationType::Attacking );
+                } else {
+                    // Currently we can't automatically generate dying animation
+                    //   because that animation type does not provide all
+                    //   8 directions in our tga files.
+                    // We need to modify eg. unit_human_peasant.def
+                    //   to repeat 4-direction animations in all
+                    //   8 directions
```

Comment:
Alternatively we could do one of the following:
1) Modify `UnitAnimation::getCurrentSpriteIndex` to add a special case for the dying animation - but that might be a little less elegant.
1) Always set the Unit's facing on death to one of the permitted directions.